### PR TITLE
have consistent null checks and returns

### PR DESCRIPTION
**Pull request by jpcima**
#172 

Harmonize the way how are performed the null checks in the public functions, and some other things.

### CODE REVIEW

File: src/adlmidi.cpp [hunk 1]
```diff
@@ -60,7 +60,7 @@ ADLMIDI_EXPORT struct ADL_MIDIPlayer *adl_init(long sample_rate)
         return NULL;
     }
 
-    MIDIplay *player = new MIDIplay(static_cast<unsigned long>(sample_rate));
+    MIDIplay *player = new(std::nothrow) MIDIplay(static_cast<unsigned long>(sample_rate));
```
Comment (Wohlstand) [inline src/adlmidi.cpp]: Will this `(std::nothrow)` compatible to C++98?

Comment (jpcima) [reply src/adlmidi.cpp]: It doesn't have a (C++11) mark on cpp reference so I assume, yes.
It has built on my djgpp copy. Now that remains to check if Watcom will accept.

Comment (Wohlstand) [reply src/adlmidi.cpp]: The bottleneck for now is an official Android NDK where the C++98 mode is default one.

Comment (Wohlstand) [reply src/adlmidi.cpp]: Oh, what if I'll add the build check on Android NDK too?

Comment (Wohlstand) [reply src/adlmidi.cpp]: Okay, I'll merge the stuff, but later I'll make the CI build for Android to verify the build on it also. It's will be better than I'll do this manually which I doing update of my Android apps not so offten.

Comment (jpcima) [reply src/adlmidi.cpp]: I could as well have kept `new` the way like it is, but remove another null check down below.
Having new and a null check together just don't make sense.

**Pull request merged**